File: ValidatorAngajat.java
package me.validator;

import me.exceptii.ValidationException;
import me.model.Angajat;

public class ValidatorAngajat implements Validator<Angajat> {
    public void validate(Angajat a) throws ValidationException {
        String erori = "";
        if(a==null)
            throw new ValidationException("Obiectul nu exista!\n");
        if(a.getNume()==null  || a.getId()==null || a.getParola()==null || a.getUsername()==null)
            throw new ValidationException("Campurile nu trebuie sa fie null");
        if(a.getNume().equals(""))
            erori+="Numele nu poate fi gol!\n";
        if(a.getVarsta()<0)
            erori+="Varsta nu poate fi mai mica un numar negativ!\n";
        if(a.getUsername().equals(""))
            erori+="Username-ul nu poate fi gol!\n";
        if(a.getParola().equals(""))
            erori+="Parola nu poate fi goala!\n";
        if(!erori.equals(""))
            throw new ValidationException(erori);
    }
}

File: Concurs Inot C#/ValidatorAngajat.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Problema_3_CSharp.me.model;
using Problema_3_CSharp.me.exceptii;

namespace Problema_3_CSharp.me.validator
{
    public class ValidatorAngajat : IValidator<Angajat>
    {
        public void Validate(Angajat a)
        {
            String erori = "";
            if (a == null)
                throw new ValidationException("Obiectul nu poate fi null!\n");
            if (a.Nume == null || a.Username == null || a.Parola == null)
                throw new ValidationException("Campurile nu pot fi null!\n");
            if (a.Nume.Equals(""))
                erori += "Numele nu poate fi gol!\n";
            if (a.Varsta < 0)
                erori += "Varsta nu poate fi negativa!\n";
            if (a.Username.Equals(""))
                erori += "Username-ul nu poate fi gol!\n";
            if (a.Parola.Equals(""))
                erori += "Parola nu poate fi goala!\n";
            if (!erori.Equals(""))
                throw new ValidationException(erori);
        }
    }
}
